src/operation.rs

```diff
@@ -2,18 +2,54 @@ use ndarray::{array, Array2};
 use num::Complex;
 use std::{f64::consts, vec};
 
+use crate::math::real_to_complex;
+
 // Naming?
 pub trait OperationTrait {
     fn matrix(&self) -> Array2<Complex<f64>>;
     fn targets(&self) -> Vec<usize>;
     fn arity(&self) -> usize;
 }
 
+/// A quantum computer operation represented by a matrix and targets.
+///
+/// - `matrix` corresponds to the quantum operator
+/// - `targets` corresponds to the operator operands
 #[derive(Clone, Debug)]
 pub struct Operation {
     matrix: Array2<Complex<f64>>,
     targets: Vec<usize>,
-    arity: usize,
+}
+
+impl Operation {
+    /// Constructs an operation with the given matrix, targets and arity.
+    ///
+    /// Returns an operation if:
+    /// - `matrix` is square with sides equal to number of `targets`
+    /// - number of `targets` is equal to `arity`
```

Comment:
Arity doesn't need to exist as a parameter here anymore

Reply:
you do not need to exist anymore!!!!!!!

Reply:
lmao xd